com.microsoft.mrtk.input/Visualizers/RiggedHandVisualizer/RiggedHandMeshVisualizer.cs

```diff
@@ -0,0 +1,302 @@
+            }
+
+            if (displaySubsystems.Count == 0)
+            {
+                SubsystemManager.GetSubsystems(displaySubsystems);
+            }
+
+            // Are we running on an XR display and it happens to be transparent?
+            // Probably shouldn't be showing rigged hands!
+            if (displaySubsystems.Count > 0 &&
+                displaySubsystems[0].running &&
+                !displaySubsystems[0].displayOpaque &&
+                !showHandsOnTransparentDisplays)
+            {
+                handRenderer.enabled = false;
+                return;
+            }
+            
+            transform.localScale = new Vector3(handNode == XRNode.LeftHand ? -handScale : handScale, handScale, handScale);
+
+            RenderMesh(joints);
+        }
+
+        private void RenderMesh(IReadOnlyList<HandJointPose> joints)
+        {
+            // Enable the hand mesh after once we have joint data
+            handRenderer.enabled = true;
+
+            float error = 0.0f;
+
+            for (int i = 0; i < joints.Count; i++)
+            {
+                HandJointPose jointPose = joints[i];
+
+                // The actual, physical, rigged joint on the armature.
+                // This actually corresponds to the "base" of the bone;
+                // as an example, riggedVisualJointsArray[IndexMetacarpal] actually
+                // corresponds to a transform that is located at the wrist joint,
+                // but points towards the metacarpal joint location.
+                // This discrepancy is because OpenXR uses joint locations/rotations,
+                // whereas armatures/Unity/Blender use *bones*.
+                Transform jointTransform = riggedVisualJointsArray[i];
+
+                if (jointTransform != null)
+                {
+                    
+                    switch ((TrackedHandJoint)i)
+                    {
+                        case TrackedHandJoint.Palm:
+                            // Don't track the palm. The hand mesh shouldn't have a "palm bone".
+                            break;
+                        case TrackedHandJoint.Wrist:
+                            // Set the wrist directly from the joint data.
+                            jointTransform.position = jointPose.Position;
+                            jointTransform.rotation = jointPose.Rotation;
+                            break;
+                        case TrackedHandJoint.ThumbTip:
+                        case TrackedHandJoint.IndexTip:
+                        case TrackedHandJoint.MiddleTip:
+                        case TrackedHandJoint.RingTip:
+                        case TrackedHandJoint.LittleTip:
+                            // The tip bone uses the joint rotation directly.
+                            jointTransform.rotation = joints[i-1].Rotation;
+                            error += Vector3.Dot((jointTransform.position - joints[i-1].Position), jointTransform.forward);
```

Comment:
Can we box this into its own function to make it clearer that the error is calculated by taking a delta between a digit's tip and its previous joint.

Reply:
Resolved